vscode/npm-package/scripts/generate-readme.js

```diff
@@ -0,0 +1,121 @@
+const fs = require("fs");
+const path = require("path");
+const ts = require("typescript");
+
+const libDir = path.resolve(__dirname, "../lib");
+const readmePath = path.resolve(__dirname, "../readme.md");
+
+let markdown = `# @microsoft/kiota
+
+This library provides various functions to interact with Kiota, a client generator for HTTP REST APIs described by OpenAPI.
+
+## Installation
+
+To install the package, use the following command:
+
+\`\`\`bash
+npm install @microsoft/kiota
+\`\`\`
+
+## Usage
+
+Once installed, you can use the available functions to generate and interact with Kiota clients.
+Below is a reference for each function, including parameters and return values.
+
+`;
+
+let tsFilePaths;
+try {
+  tsFilePaths = fs
+    .readdirSync(libDir)
+    .filter((file) => file.endsWith(".ts"))
+    .map((file) => path.join(libDir, file));
+} catch (err) {
+  console.error(`Error reading directory: ${err.message}`);
+  process.exit(1);
+}
+
+for (const filePath of tsFilePaths) {
+
+  let fileContent;
+  try {
+    fileContent = fs.readFileSync(filePath, "utf-8");
+  } catch (err) {
+    console.error(`Error reading file ${filePath}: ${err.message}`);
+    continue;
+  }
+
+  const sourceFile = ts.createSourceFile(filePath, fileContent, ts.ScriptTarget.Latest, true);
```

Comment:
🍾. Very insightful use of the ts.createSourceFile function. Will ensure quick generation of useful context and uniform pattern in the output. Question, can I  use the `param_name=<value>` syntax without wrapping it into an options object to the functions or I have to pass them nested in an object i.e `{param_name: <value>} ` syntax. If the former is acceptable, then do the param names have to start with `option.` prefix e.g.  the `config.pluginName` instead of just `pluginName`?

Reply:
I don't thin that they have to start with the prefix. It was how the jsdoc was defined that gets the response in that way. To show the end user that it's going to be part of an object at the end of the day